staging/src/k8s.io/apiserver/pkg/endpoints/handlers/finisher/finisher.go

```diff
@@ -104,6 +116,43 @@ func FinishRequest(ctx context.Context, fn ResultFunc) (runtime.Object, error) {
 	case result := <-resultCh:
 		return result.Return()
 	case <-ctx.Done():
-		return nil, errors.NewTimeoutError(fmt.Sprintf("request did not complete within requested timeout %s", ctx.Err()), 0)
+		// we are going to send a timeout response to the caller, but the asynchronous goroutine is
+		// still executing the ResultFunc function.
+		// kick off a go routine here to wait for the ResultFunc to return and then log some details.
+		defer func() {
+			go func() {
+				startedAt := time.Now()
```

Comment:
this isn't actually started, this is the time the request timedout, right?

Reply:
yes, after the request had timed out, this the `startedAt` for the post-timeout tracker. `timedOutAt` reads better?

Reply:
@deads2k I renamed to `timedOutAt`.